dnf/conf/config.py

```diff
@@ -243,6 +245,32 @@ def get_reposdir(self):
             dnf.util.ensure_dir(myrepodir)
         return myrepodir
 
+    def _check_remote_file(self, optname):
+        """
+        In case the option value is a remote URL, download it to the temporary location
+        and use this temporary file instead.
+        """
+        prio = self._get_priority(optname)
+        val = self._get_value(optname)
+        if isinstance(val, basestring):
+            location = urlparse.urlparse(val)
+            if location[0] in ('file', ''):
+                # just strip the file:// prefix
+                self._set_value(optname, location.path, prio)
+            else:
+                downloader = libdnf.repo.Downloader()
+                temp_fd, temp_path = tempfile.mkstemp()
```

Comment:
I think it would be a good idea to add a prefix to the file, along the lines of `"dnf-config-"` or `"dnf-downloaded-config-"`, just in case a user ever bumps into it...

Reply:
Good idea.